src/main/scala/scalismo/mesh/boundingSpheres/ClosestPoint.scala

```diff
@@ -36,9 +36,9 @@ case class ClosestPoint(point: Point[_3D], distanceSquared: Double) {
  * @param point The closest point location on the surface.
  * @param distanceSquared The squared distance to the closest point location.
  */
-sealed abstract class ClosestPointOnSurface(val point: Point[_3D], val distanceSquared: Double) {
+sealed abstract class ClosestPointWithSquaredDistance(val point: Point[_3D], val distanceSquared: Double) {
```

Comment:
The `ClosestPoint` class is for the most performant queries where no meta-information (isVertex, isOnLine, isInTriangle etc.) is needed. As it is a case class, it can not serve as the base class for the other closest point classes with more information. `ClosestPointWithMetaInformation` or `ClosestPointBase` would be a more appropriate name. But I do not like those names. Somehow the name should make it clear that one can get more information when one matches its concrete type.

Reply:
I now changed the name to `ClosestPointSpecialized`. While I am still not totally happy with that name, I think it comes closest to what it should express.

Further, I added the method `distance` that returns the square root of the `squaredDistance`.